src/test/java/simplifier/UserControllerIT.java

```diff
@@ -16,37 +12,59 @@
 import simplifier.model.User;
 import simplifier.repositories.UserRepository;
 
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertThat;
+
 @RunWith(SpringRunner.class)
-@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT )
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 public class UserControllerIT {
 
-  private TestRestTemplate restTemplate = new TestRestTemplate();
+    private TestRestTemplate restTemplate = new TestRestTemplate();
+
+    private UserRepository userRepository;
+
+    @Autowired
+    public void setUserRepository(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @Autowired
+    public void setRestTemplate(TestRestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    @Test
+    public void createUser() {
+        User user = new User();
+        user.setUsername("test");
+        user.setPassword("test");
+
+        ResponseEntity<User> responseEntity = restTemplate.postForEntity("/user",
+                user, User.class);
 
-  private UserRepository userRepository;
+        User savedUser = userRepository.findByUsername("test");
 
-  @Autowired
-  public void setUserRepository(UserRepository userRepository) {
-    this.userRepository = userRepository;
-  }
+        assertThat(savedUser, is(user));
+        assertThat(responseEntity.getStatusCode(), is(HttpStatus.CREATED));
+        assertThat(responseEntity.getBody(), is(nullValue()));
+    }
 
-  @Autowired
-  public void setRestTemplate(TestRestTemplate restTemplate) {
-    this.restTemplate = restTemplate;
-  }
+    @Test
+    public void conflictIfUsernameNotUnique() {
+        User firstUser = new User();
+        firstUser.setUsername("test");
+        firstUser.setPassword("test");
 
-  @Test
-  public void createUser() {
-   User user = new User();
-   user.setUsername("test");
-   user.setPassword("test");
+        userRepository.save(firstUser);
 
-   ResponseEntity<User> responseEntity = restTemplate.postForEntity("/user",
-       user, User.class);
+        User secondUser = new User();
+        secondUser.setUsername("test");
+        secondUser.setPassword("test");
 
-   User savedUser = userRepository.findByUsername("test");
+        ResponseEntity<String> responseEntity = restTemplate.postForEntity("/user",
+                secondUser, String.class);
 
-   assertThat(savedUser, is(user));
-   assertThat(responseEntity.getStatusCode(), is(HttpStatus.CREATED));
-   assertThat(responseEntity.getBody(), is(nullValue()));
-  }
+        assertThat(responseEntity.getStatusCode(), is(HttpStatus.CONFLICT));
```

Comment:
Verifying that only one user is saved and response body has message are required